consensus/abci.go

```diff
@@ -70,6 +70,11 @@ func NewTendermintABCI(db *database.Database, ethRPCClient *rpc.Client, metrics
 		metrics:           metrics,
 	}
 
+	err = abci.ResetBlockState()
```

Comment:
Why do you reset current state as part of initializing a new instance? IMO That logic should be restored as part of initNode()

Reply:
Because it doesn't belong to initNode. It's part of initializing of ABCI struct. If u check further what it does, it only setups zero values for the temporary state that belongs to ABCI and is modified between BeginBlock and EndBlock. To me it makes sense because ABCI is not a valid struct till the moment this method is called.